eio/io.ml

```diff
@@ -0,0 +1,181 @@
+open Websocket
+open Astring
+open Eio
+
+type mode = Client of (int -> string) | Server
+
+let is_client mode = mode <> Server
+
+let xor mask msg =
+  for i = 0 to Bytes.length msg - 1 do
+    (* masking msg to send *)
+    Bytes.set msg i
+      Char.(to_int mask.[i mod 4] lxor to_int (Bytes.get msg i) |> of_byte)
+  done
+
+let is_bit_set idx v = (v lsr idx) land 1 = 1
+let set_bit v idx b = if b then v lor (1 lsl idx) else v land lnot (1 lsl idx)
+let int_value shift len v = (v lsr shift) land ((1 lsl len) - 1)
+
+(* Eio only uses Cstructs so we lose 0copy :(! *)
+let read_exactly src remaining =
```

Comment:
I think this function is just `Buf_read.take` (except for return an option instead of `End_of_file`).

Reply:
Thanks :))